### PR TITLE
dviread: find_tex_file: Ensure the encoding on windows

### PR DESCRIPTION
**Pull request by Kojoley**
ref #11848 and #12253

### CODE REVIEW

File: lib/matplotlib/dviread.py [hunk 1]
```diff
@@ -1013,17 +1013,23 @@ def find_tex_file(filename, format=None):
...
+        kwargs = dict(env=dict(os.environ, command_line_encoding='utf-8'))
+    else:
+        kwargs = {}
+
     cmd = ['kpsewhich']
     if format is not None:
         cmd += ['--format=' + format]
     cmd += [filename]
     try:  # Below: strip final newline.
-        result = cbook._check_and_log_subprocess(cmd, _log)[:-1]
```
Comment (jklymak) [inline lib/matplotlib/dviread.py]: #12356 and this differ here....  Can that PR and this one be combined?

Comment (anntzer) [reply lib/matplotlib/dviread.py]: @Kojoley Can you include the fix from #12356 into this PR?

Comment (Kojoley) [reply lib/matplotlib/dviread.py]: Yes, should I cherry pick your commit into my PR?

Comment (anntzer) [reply lib/matplotlib/dviread.py]: cherry-pick or just reproduce it, either way is fine.

**Comment by jklymak** [comment]
Leaving open a bit so @anntzer can double check his changes (look right to me)

Not sure if this one could have a test or not...

**Comment by anntzer** [comment]
tests means: we should reenable tex tests on appveyor...

**Comment by Kojoley** [comment]
For some reason pytest hangs if `TEST_ALL: "yes"`...

**Pull request merged**